aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -988,3 +1011,107 @@ impl AsMut<AptosVMImpl> for AptosVM {
         &mut self.0
     }
 }
+
+impl AptosSimulationVM {
+    fn validate_simulated_transaction<S: MoveResolverExt>(
+        &self,
+        session: &mut SessionExt<S>,
+        transaction: &SignedTransaction,
+        txn_data: &TransactionMetadata,
+        allow_too_new: bool,
+        log_context: &AdapterLogSchema,
+    ) -> Result<(), VMStatus> {
+        self.0.check_transaction_format(transaction)?;
+
+        let prologue_status =
+            self.0
+                .run_prologue_with_payload(session, transaction.payload(), txn_data, log_context);
+        match prologue_status {
+            Err(err)
+                if !allow_too_new || err.status_code() != StatusCode::SEQUENCE_NUMBER_TOO_NEW =>
+            {
```

Comment:
this feels like it would be more efficient as an if / let -- but I don't think we need it just yet because allow_to_new isn't really exposed.